src/backend/catalog/heap.c

```diff
@@ -1283,6 +1303,12 @@ heap_create_with_catalog(const char *relname,
 	else
 		relacl = NULL;
 
+	if (relrewrite != InvalidOid)
+	{
+		old_rel = table_open(relrewrite, AccessShareLock);
```

Comment:
can't we close the table immediately here?

Reply:
old_rel is a poiter to the shared buffer. We could do memcpy here but I think it is safer just to hold the lock while we are "referring" to the rel. The parallel process might for example delete this relation and then we may get in to trouble in smgrcteate